src/volatile_memory.rs

```diff
@@ -290,6 +290,37 @@ impl<'a> VolatileSlice<'a> {
         self.size == 0
     }
 
+    /// Divides one slice into two at an index.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use vm_memory::VolatileMemory;
+    /// let mut mem = [0u8; 32];
+    /// let mem_ref = &mut mem[..];
+    /// let vslice = mem_ref.get_slice(0, 32).unwrap();
+    /// let (start, end) = vslice.split_at(8).unwrap();
+    /// assert_eq!(start.len(), 8);
+    /// assert_eq!(end.len(), 24);
+    /// let (start, end) = vslice.split_at(0).unwrap();
+    /// # assert_eq!(start.len(), 0);
+    /// # assert_eq!(end.len(), 32);
+    /// # let (start, end) = vslice.split_at(31).unwrap();
+    /// # assert_eq!(start.len(), 31);
+    /// # assert_eq!(end.len(), 1);
+    /// # let (start, end) = vslice.split_at(32).unwrap();
+    /// # assert_eq!(start.len(), 32);
+    /// # assert_eq!(end.len(), 0);
```

Comment:
Do you mind moving the checks to the unit tests and keeping just a meaningful example here?

It looks a bit cluttered when opening the documentation with `cargo doc --open`, and it's not really IDE compatible (you cannot run them via something like Intelij easily).

Reply:
sure, although the # lines shouldn't be visible in the generated doc by default.